tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/transform/OperationContextCppCodeGenerator.java

```diff
@@ -41,7 +41,17 @@ public void addVariableInScope(String var)
         }
         this.varName.push(Pair.of(var, false));
     }
- 
+
+    public void cleanupVariablesCurrentScope() {
+      while (!this.getVarName().isEmpty()) {
+        if (!this.getVarName().peek().right) {
+          this.getVarName().pop();
+        } else {
+          break;
```

Comment:
would rather have this handled in the loop than a break statement
```java
while (!this.getVarName().isEmpty() && !this.getVarName().peek().right) {
  this.getVarName().pop();
}
```

Reply:
shall change